web/api/v1/api.go

```diff
@@ -780,9 +784,16 @@ func getGlobalURL(u *url.URL, opts GlobalURLOptions) (*url.URL, error) {
 				// this port, but it's still the best possible guess.
 				host, _, err := net.SplitHostPort(opts.Host)
 				if err != nil {
-					return u, err
+					if strings.HasSuffix(err.Error(), "missing port in address") {
+						host = opts.Host
+					} else {
+						return u, err
+					}
+				}
+				u.Host = host
+				if port != "" {
+					u.Host += ":" + port
```

Comment:
You probably want to use https://golang.org/pkg/net/?m=all#JoinHostPort here so that IPv6 addresses are handled correctly as well? And then add test cases below where the global URL contains a IPv6 addresses (with and without port).

Reply:
Indeed  . I did think about then forget :) Thanks, will update the PR

Reply:
@juliusv it turns out that in a URL, there should be [] for IPv6. Therefore, we are already covered.

Adding JoinHostPort  creates a URL like `[[::1]]:80` which is not valid, therefore the current approach is correct.